src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCIContainerService.java

```diff
@@ -366,17 +359,20 @@ public Path createBuildScript(ProgrammingExerciseParticipation participation) {
                 cd /testing-dir
                 """);
 
-        if (actions != null) {
-            actions.forEach(action -> buildScript.append(action.getScript()).append("\n"));
-        }
-        else {
+        actions.forEach(action -> buildScript.append(action.getScript()).append("\n"));
+
+        // Fall back to hardcoded scripts for old exercises without windfile
```

Comment:
Instead of adding this comment here I would prefer if this was moved to a method

e.g. `handleExerciseWithoutWindfileConfiguration` and add the javadoc there

Reply:
I think I will leave it like for now since it will be removed soon anyways